### PR TITLE
KAN-66-refactor-everything-to-use-shadcn

### PR DESCRIPTION
**Pull request by henriquesbrissia**

### CODE REVIEW

File: imports/ui/components/PasswordUpdate.tsx [hunk 1]
```diff
@@ -31,29 +35,79 @@ export const PasswordUpdate = () => {
     });
   };
 
+  const form = useForm();
```
Comment (henriquealbert) [inline imports/ui/components/PasswordUpdate.tsx:38]: why? you already have useForm declared in line 18

File: imports/ui/components/PasswordUpdate.tsx [hunk 2]
```diff
@@ -31,29 +35,79 @@ export const PasswordUpdate = () => {
...
+    <>
+      <div className="bg-slate-100 p-8 rounded-md shadow-md">
+        <h1 className="text-2xl font-bold mb-4">Change password</h1>
+        <p className="text-sm text-gray-500">
+          Ensure your account is using a long, random password to stay secure.
+        </p>
+      </div>
+      <div className="bg-white p-8 rounded-md shadow-sm">
+        <Form {...form}>
+          <form onSubmit={handleSubmit(onSubmit)} className="space-y-6">
```
Comment (henriquealbert) [inline imports/ui/components/PasswordUpdate.tsx:50]: you already have Form component, no need to add another form tag here, use the Form component

File: imports/ui/pages/Profile.tsx [hunk 3]
```diff
@@ -83,42 +83,84 @@ export const Profile = () => {
 
   const profilePicture = watch("image");
 
+  const form = useForm();
```
Comment (henriquealbert) [inline imports/ui/pages/Profile.tsx]: same problem here

File: imports/ui/pages/Profile.tsx [hunk 4]
```diff
@@ -83,42 +83,84 @@ export const Profile = () => {
...
-              alt="Profile Picture"
-              style={{ width: "150px", height: "150px", borderRadius: "50%" }}
+    <div className="grid gap-4 grid-cols-2 grid-rows-4 max-w-4xl mx-auto p-8">
+      <div className="bg-white p-8 rounded-md shadow-sm">
+        <h1 className="text-2xl font-bold mb-4">Your Account</h1>
+        <p className="text-sm text-gray-500">Update your account's name and email address.</p>
+      </div>
+      <div className="bg-slate-100 p-8 rounded-md shadow-md">
+        <Form {...form}>
+          <form onSubmit={handleSubmit(handleUpdateProfile)} className="space-y-6">
```
Comment (henriquealbert) [inline imports/ui/pages/Profile.tsx]: same problem here

File: imports/ui/pages/Profile.tsx [hunk 5]
```diff
@@ -83,42 +83,84 @@ export const Profile = () => {
...
+            <div className="flex items-center gap-4">
+              {profilePicture ? (
+                <img
+                  src={profilePicture}
+                  alt="Profile"
+                  className="w-32 h-32 rounded-full bg-gray-200 border"
+                />
+              ) : (
+                <div className="w-16 h-16 rounded-full bg-gray-200 flex items-center justify-center text-gray-400">
+                  H
```
Comment (henriquealbert) [inline imports/ui/pages/Profile.tsx]: what happens if my user starts with Z? create a function that will handle this logic

File: imports/ui/pages/Profile.tsx [hunk 6]
```diff
@@ -83,42 +83,84 @@ export const Profile = () => {
...
-      <button onClick={handleDeleteAccount} style={{ marginTop: "20px", color: "red" }}>
-        Delete Account
-      </button>
+            <Button type="submit" className="w-24">
+              {isSubmitting ? "Saving..." : "Save"}
+            </Button>
+          </form>
+        </Form>
+      </div>
+      {user?.services?.password && <PasswordUpdate />}
```
Comment (henriquealbert) [inline imports/ui/pages/Profile.tsx]: dont expose the password in the front end, this is insecure, create a logic in the backend, for instance: withPassword/hasPassword: boolean, then use it here

**Review by henriquealbert — CHANGES_REQUESTED** [review]

File: imports/ui/pages/Profile.tsx (hunk 5 above)
Comment (henriquesbrissia) [reply imports/ui/pages/Profile.tsx]: Nice catch, fixed already.
Thank you!

File: imports/api/users/methods.ts [hunk 7]
```diff
@@ -20,7 +20,16 @@ export const usersModule = createModule("users")
...
     }
     const user = await UsersCollection.findOneAsync({ _id: userId });
-    return user;
+    return {
+      _id: user?._id,
+      hasPassword: !!user?.services?.password,
+      profile: user?.profile,
+      emails: user?.emails,
+      githubMail: user?.services?.github?.email,
+      googleMail: user?.services?.google?.email,
```
Comment (henriquealbert) [inline imports/api/users/methods.ts]:
```suggestion
      githubEmail: user?.services?.github?.email,
      googleEmail: user?.services?.google?.email,
```

File: imports/ui/pages/Profile.tsx [hunk 8]
```diff
@@ -1,38 +1,28 @@
...
-        user?.services?.github?.email ||
-        user?.services?.google?.email ||
-        user?.emails[0].address ||
-        "",
-      image:
-        user?.services?.github?.avatar ||
-        user?.services?.google?.picture ||
-        user?.profile?.image ||
-        ""
+      email: user?.githubMail || user?.googleMail || user?.emails?.[0].address || "",
```
Comment (henriquealbert) [inline imports/ui/pages/Profile.tsx]:
```suggestion
      email: user?.githubEmail || user?.googleEmail || user?.emails?.[0].address || "",
```

**Pull request merged**